src/main/java/org/terasology/climateConditions/TemperatureSystem.java

```diff
@@ -0,0 +1,99 @@
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.logic.players.event.OnPlayerSpawnedEvent;
+import org.terasology.registry.CoreRegistry;
+import org.terasology.registry.In;
+import org.terasology.world.generation.GeneratingRegion;
+
+
+@RegisterSystem(value = RegisterMode.AUTHORITY)
+public class TemperatureSystem extends BaseComponentSystem {
+    @In
+    BiomeRegistry biomeRegistry;
+    @In
+    CoreRegistry coreRegistry;
+    @In
+    GeneratingRegion generatingRegion;
+    @In
+    private Time time;
+    @In
+    private EntityManager entityManager;
+    @In
+    ClimateConditionsSystem climateConditionsSystem;
+    @In
+    DelayManager delayManager;
+
+    private static final int CHECK_INTERVAL = 1000;
+
+    private static final int DISPLAY_INTERVAL = 10000;
+
+
+    private float bodyTemperature;
+
+    private float lastBodyTemperature;
+
+    public static final String BODY_TEMPERATURE_UPDATE_ACTION_ID = "Body Temperature Update";
+
+    public static final String BODY_TEMPERATURE_DISPLAY_ACTION_ID = "Body Temperature Display";
+
+    @ReceiveEvent
+    public void onSpawn(OnPlayerSpawnedEvent event, EntityRef player) {
+        bodyTemperature = 0.4f;
+        lastBodyTemperature = 0.4f;
+        delayManager.addPeriodicAction(player, BODY_TEMPERATURE_UPDATE_ACTION_ID, 0, CHECK_INTERVAL);
+        delayManager.addPeriodicAction(player, BODY_TEMPERATURE_DISPLAY_ACTION_ID, 0, DISPLAY_INTERVAL);
+
+    }
+
+    @ReceiveEvent
+    public void onTemperatureUpdate(PeriodicActionTriggeredEvent event, EntityRef player, LocationComponent location) {
+        if (event.getActionId().equals(BODY_TEMPERATURE_UPDATE_ACTION_ID)) {
+            float envTemperature = climateConditionsSystem.getTemperature(location.getLocalPosition().getX(),
+                    location.getLocalPosition().getY(), location.getLocalPosition().getZ());
+            float envHumidity = climateConditionsSystem.getHumidity(location.getLocalPosition().getX(),
+                    location.getLocalPosition().getY(), location.getLocalPosition().getZ());
+            bodyTemperature =
+                    lastBodyTemperature + ((((envTemperature - (envHumidity / 10)) - lastBodyTemperature) / 100000) * CHECK_INTERVAL);
+            lastBodyTemperature = bodyTemperature;
+        }
+    }
+
+    @ReceiveEvent
+    public void onTemperatureDisplay(PeriodicActionTriggeredEvent event, EntityRef player, LocationComponent location) {
+        if (event.getActionId().equals(BODY_TEMPERATURE_DISPLAY_ACTION_ID)) {
+            float envTemperature = climateConditionsSystem.getTemperature(location.getLocalPosition().getX(),
+                    location.getLocalPosition().getY(), location.getLocalPosition().getZ());
+            player.getOwner().send(new ChatMessageEvent("Body Temperature: " + bodyTemperature, player.getOwner()));
+            player.getOwner().send(new ChatMessageEvent("Env Temperature: " + envTemperature, player.getOwner()));
```

Comment:
If having that information in the in-game text field helps you during development, that's fine, but eventually this should - if at all - be a (debug) log

Reply:
Yes obviously, I figured you might also want to take a look at the rate of change of temperature (too fast / too slow etc..) It helps while running around to get an idea of the temperature at various points in the world and well again the rate at which the temperature is changing.